src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -421,9 +421,28 @@ void GCToOSInterface::FlushProcessWriteBuffers()
         // Iterate through each of the threads in the list.
         for (mach_msg_type_number_t i = 0; i < cThreads; i++)
         {
-            // Request the threads pointer values to force the thread to emit a memory barrier
-            size_t registers = 128;
-            machret = thread_get_register_pointer_values(pThreads[i], &sp, &registers, registerValues);
+            if (__builtin_available (macOS 10.14, iOS 12, tvOS 9, *))
+            {
+                // Request the threads pointer values to force the thread to emit a memory barrier
+                size_t registers = 128;
+                machret = thread_get_register_pointer_values(pThreads[i], &sp, &registers, registerValues);
+            }
+            else
+            {
+                // fallback implementation for older OS versions
```

Comment:
We only really need an implementation for arm64 since I just saw that newest Xcode doesn't support iOS 11 simulators anymore which is the only x64 iOS host.

Reply:
Ok, so this is fine. The reason why Apple recommended using the `thread_get_register_pointer_values` was to make it architecture agnostic. It is just a wrapper for thread_get_state.